Split/Network/Sync/FeatureFlagsSynchronizer.swift

```diff
@@ -148,36 +151,45 @@ class DefaultFeatureFlagsSynchronizer: FeatureFlagsSynchronizer {
     private func filterSplitsInCache() {
         let splitsStorage = storageContainer.persistentSplitsStorage
         let currentSplitsQueryString = splitsFilterQueryString
-        if currentSplitsQueryString == splitsStorage.getFilterQueryString() {
-            return
-        }
-
-        let filters = splitConfig.sync.filters
-        let namesToKeep = Set(filters.filter { $0.type == .byName }.flatMap { $0.values })
-        let prefixesToKeep = Set(filters.filter { $0.type == .byPrefix }.flatMap { $0.values })
-        let setsToKeep = Set(filters.filter { $0.type == .bySet }.flatMap { $0.values })
-
-        let splitsInCache = splitsStorage.getAll()
-        var toDelete = [String]()
 
-        for split in splitsInCache {
-            guard let splitName = split.name else {
-                continue
+        let filterHasChanged = currentSplitsQueryString != splitsStorage.getFilterQueryString()
+        let flagsSpecHasChanged = flagsSpec != splitsStorage.getFlagsSpec()
+
+        // if neither the filter nor the flags spec have changed, we don't need to do anything
+        if filterHasChanged || flagsSpecHasChanged {
+            let splitsInCache = splitsStorage.getAll()
+            var toDelete = [String]()
+
+            if flagsSpecHasChanged {
+                // when flagsSpec has changed, we delete everything
+                toDelete = splitsInCache.compactMap { $0.name }
+            } else if filterHasChanged {
```

Comment:
```
} else if filterHasChanged {
```
or 
```
}  
if filterHasChanged {
``` ?
(in case both `filterHasChanged` and `flagsSpecHasChanged` are true)

Reply:
In this case, if filter has changed, stored flags are filtered. Since when spec changes all flags are removed, there's no point in filtering